lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -2122,14 +2122,15 @@ class EngineSemanticsOwner {
     // First, update each object's information about itself. This information is
     // later used to fix the parent-child and sibling relationships between
     // objects.
-    for (final SemanticsNodeUpdate nodeUpdate in update._nodeUpdates!) {
+    final List<SemanticsNodeUpdate> nodeUpdates = update._nodeUpdates!;
```

Comment:
please add an assert where the `nodeUpdates` variable goes out of scope that verifies that `update._nodeUpdates` has not changed

Reply:
Done